pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandlerTest.java

```diff
@@ -1374,7 +1373,7 @@ public void testDisableForwardIndexForSingleRawColumn()
         // This value is based on the rows in createTestData().
         dictionaryElementSize = 7;
       } else if (dataType == FieldSpec.DataType.BIG_DECIMAL) {
-        dictionaryElementSize = 11;
+        dictionaryElementSize = 4;
```

Comment:
Not sure I follow the root cause ? Can you elaborate why changing this fixes flakeyness ?

Reply:
The existing code uses a random BASE_BIG_DECIMAL. Depending on the random value, the dictionaryElementSize varies.  
In the `validateMetadata()`, we check if the dictionaryElementSize remains the same
`assertEquals(columnMetadata.getColumnMaxLength(), dictionaryElementSize, column);`

This was causing the flakiness. Now we make it deterministic by not using BASE_BIG_DECIMAL.